### ISSUE TITLE
How to use join with like automapping?

### ISSUE THREAD

**Issue by robbelroot** [sqlkata/querybuilder#424]
When i use a join (which works as i can see when i look at the non generic function return value) with the generic overload GetAsync<Entity>(), then the relational properties are still null. Am i missing something or isn't it possible with sqlkata?

**Comment by ahmad-moussawi** [comment]
@robbelroot could you provide a code example?

**Comment by robbelroot** [comment]
Sure, but it's just a trivial join, nothing special:
`_db.Query($"Jobs")
                .Join($"Customers", "Jobs.customer_id", "Customers..id")`

**Comment by ahmad-moussawi** [comment]
No, it doesn't work this way, the joined columns will be returned on the same level as the original table record.
For example, the result class should be similar to:

```cs
public class JobWithCustomer {
 public int JobId;
 public string Position;
 public int CustomerId;
 public string CustomerName;
}
```
for a query like this one
```cs
db.Query("Jobs").Join("Customers", ...).Select("Jobs.{JobId, Position}", "Customers.{CustomerId, CustomerName}");
```

You can use the `.Include` and `.IncludeMany` to include related properties, but the only limitation, for now, is that this will work only on the non-typed overload of `Get` and `GetAsync`.

**Comment by robbelroot** [comment]
`_db.Query("job").Include("customer", q => q.From("customer"))`

Gives me the error:

> Lambda expression cannot be converted to type Query because it's no delegate type.
